### DISCUSSION TITLE
How to run a script before deployment on a monorepo?

### DISCUSSION THREAD

**Discussion by robertt** [vercel/vercel#3914] [General]
So I have an `nx` managed monorepo, and so far this configuration has worked great: 
```
{
  "version": 2,
  "builds": [
    { "src": "/dist/apps/api/main.js", "use": "@now/node" },
    { "src": "/dist/apps/web/*", "use": "@now/static" }
  ],
  "routes": [
    { "src": "/api/(.*)", "dest": "/dist/apps/api/main.js" },
    { "handle": "filesystem" },
    { "src": "/assets/(.*)", "dest": "/dist/apps/web/assets/$1" },
    { "src": "/(.*).(js|css|ico)", "dest": "/dist/apps/web/$1.$2" },
    { "src": "/(.*)", "dest": "/dist/apps/web/index.html" }
  ]
}
```
It works great when I have the builds already existing (the dist/apps) directory. The problem is, when using the github app it doesn't build it automatically. I was wondering if there was any way to run a command (like `npm run build`) before attempting to deploy the site.

I've been looking for a solution for hours now and I can't find one... Any help would be greatly appreciated!

**Comment by paulogdm** [comment]
Not at this moment, no.

However, I do think this will be possibly solved once we release proper mono-repo support. A few links that can assist you:

- [Project Settings](https://zeit.co/blog/advanced-project-settings)
- [Custom Build Step for Node.js](https://zeit.co/docs/runtimes#advanced-usage/advanced-node-js-usage/custom-build-step-for-node-js)

**Reply by robertt** [reply]
would it cause any problems to add a beforeAll like script? It would help a lot of people using monorepos

**Reply by paulogdm** [reply]
I really think the solution here is to wait for "monorepo support". After that, we can discuss any cases not covered and that needs further clarification.

**Reply by mcsdevv** [reply]
I can't give you an exact timeline but this is a real priority for us @robertwestbury - so I'd definitely expect to see it sooner rather than later.
